tools/c7n_azure/tests/test_cosmos_db.py

```diff
@@ -199,9 +234,68 @@ def test_replace_offer_collection_action(self):
         collections = p.run()
         self.assertEqual(len(collections), 1)
 
-        self.initial_offer = collections[0]['c7n:offer'][0]
-        self.collection = collections[0]
         self._assert_offer_throughput_equals(500, collections[0]['_self'])
+        self.offer = collections[0]['c7n:offer']
```

Comment:
I think it should happen before any assert because you rely on it to restore db state.